### ISSUE TITLE
More file/stream closing improvements

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8645]
|[<img alt="scop's avatar" src="https://avatars.githubusercontent.com/u/109152?s=50" width="50" height="50">](https://github.com/scop)| @scop reported|
|-|-|
|Trac ID|trac#8645|
|Type|enhancement|
|Created|2016-07-21 09:25:26Z|

<details><summary>Searchable metadata</summary>

```
trac-id__8645 8645
type__enhancement enhancement
reporter__scop scop
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469093126212365 1469093126212365
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="scop's avatar" src="https://avatars.githubusercontent.com/u/109152?s=50" width="50" height="50">](https://github.com/scop)<a name="note_1"></a>|@scop commented|
|-|-|

https://github.com/twisted/twisted/pull/377

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

In changeset 51e078936814062ea42360516b7398c78769b4ae
```
#!CommitTicketReference repository="" revision="51e078936814062ea42360516b7398c78769b4ae"
Add topfile for #8645
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc commented|
|-|-|

In changeset a6fcf93684c5c7db76d51c15d4e95d48917ecae9
```
#!CommitTicketReference repository="" revision="a6fcf93684c5c7db76d51c15d4e95d48917ecae9"
Add topfile for #8645
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset be3b2e95bf826a7fd328036821037d78a4efd4b9
```
#!CommitTicketReference repository="" revision="be3b2e95bf826a7fd328036821037d78a4efd4b9"
Merge pull request #389 from twisted/8645-scop-closing-2

Author: scop
Reviewer: markrwilliams, rodrigc
Fixes: #8645

More file/stream closing improvements
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
